### ISSUE TITLE
Admin login credentials incompatible with estuary-www

### ISSUE THREAD

**Issue by neelvirdy** [application-research/estuary#815]
See: https://filecoinproject.slack.com/archives/C016APFREQK/p1671516580176459

the salt and the codepath used to generate the credentials via `./estuary setup` differ from how the register user path does it on estuary-www. the /login endpoint only supports the register user path from estuary-www, so logging in as an admin user fails

**Comment by alvin-reyes** [comment]
Yeah this is because of the fixed salt used by the www and the bcrypt-go incompatibility with how bcrypt-js wraps the randomization of the salt, more specifically, bcrypto-go doesn't allow fixed salt.

My solution for this is to have a toggle on the frontend to indicate if the user is admin or not (a checkbox). If it's an admin, it'll pass the "password" over the wire (HTTP) instead of salting it cause this allows the backend to salt it using bcrypt-go in the same manner as when you created the admin account. 

if it's not admin, it uses the same process e.i salt it with a fix value and process the credentials as is.

https://github.com/application-research/estuary-www/pull/68/commits/1827ffa46a3194ad09e95a82a09921a16bdeb234
https://github.com/application-research/estuary-www/pull/66

My suggestion:
-  is either we go with my changes (reopen and merge) - **this is ok.**
- OR remove the "fixed" salt on www and unify both methods - **this is the best.**
- OR modify bcrypt-go to allow specifying a fixed salt to make it consistent with the fixed salt from the www - **I don't recommend this** simply because (1) - fixed salt is not secure and (2) - forking bcrypt-go just to change the access modifier is a stretch for us. We don't want to maintain an encryption library.

**Comment by neelvirdy** [comment]
@alvin-reyes I opened the following PRs to address this issue and https://github.com/application-research/estuary-www/issues/99, which was blocked on admin login working:
- https://github.com/application-research/estuary/pull/861
- https://github.com/application-research/estuary-www/pull/164
